### PR TITLE
:sparkles: Make providers return a builtin config so engine can look at updated locations 

### PR DESCRIPTION
**Pull request by shawn-hurley**
Some providers can mutate the original location given in init config. For example, for binaries, java provider decompiles and uses the new decompiled path. This PR adds a way for providers to communicate updated locations back to the engine so builtin provider can be configured correctly without the user having to do it.

### CODE REVIEW

File: .github/workflows/demo-testing.yml [hunk 1]
```diff
@@ -88,25 +100,31 @@ jobs:
...
+          docker push ttl.sh/konveyor-operator-bundle-${GITHUB_SHA}:2h
   e2e:
     needs: test
-    uses: konveyor/ci/.github/workflows/global-ci.yml@main
+    uses: konveyor/ci/.github/workflows/global-ci-bundle.yml@main
     with:
-      component_name: tackle2-addon-analyzer
-      api_tests_ref: "${{ needs.test.outputs.api_tests_ref }}"
+      operator_bundle: ttl.sh/konveyor-operator-bundle-${{ github.sha }}:2h
+      api_tests_ref: refs/pull/165/merge
```
Comment (pranavgaikwad) [inline .github/workflows/demo-testing.yml:130]: need to revert this back

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]

File: .github/workflows/demo-testing.yml (hunk 1 above)
Comment (shawn-hurley) [reply .github/workflows/demo-testing.yml]: updated

**Comment by aufi** [comment]
LGTM, Just a note run API tests with not-merged PR from go-konveyor-tests could be done adding "API tests PR: XYZ" to PR description (details https://github.com/konveyor/ci/blob/main/.github/workflows/global-ci.yml#L150)

**Review by djzager — APPROVED** [review]
lgtm

**Review by pranavgaikwad — APPROVED** [review]

**Pull request merged**